docs: analytics: add DVC_NO_ANALYTICS mention (#2724)
iterative/dvc.org: content/docs/user-guide/analytics.md

This will disable it for the <abbr>project</abbr>. Alternatively, you can use
the `--global` or `--system` options of `dvc config` to disable analytics for

> the active user or for everyone in the system, respectively. You could also set
> environment variable `DVC_NO_ANALYTICS` to disable analytics (will override
> config).

Reviewer: ```suggestion
the active user or for everyone in the system, respectively.

Another way to disable analytics is to set the environment variable
`DVC_NO_ANALYTICS` (will override config).

```
Not sure if we want a last new line or not 🤷

> analytics.

Reviewer 2: I have suggested to rename this as `DVC_NO_ANALYTICS` as there's no reason for the users to force-enable them. https://github.com/iterative/dvc/pull/6448#discussion_r690014104

Author: I know that this is a little extra, but it is nice to have an ability to force-enable analytics for development/debugging even if it is disabled in config. `*_NO_*` env/config options get confusing because of negation (e.g. remember `no_traverse`?)

Reviewer 2: I don't really understand the development/debugging argument at all. It avoids running that code, and by default, we force analytics, so it better to just not specify the envvar? This envvar is for users as a part of an API, solely for disabling analytics. Other debugging/development should be done by some other means.

Reviewer 2: Also, see homebrew (https://docs.brew.sh/Manpage#environment) and keepsake (https://keepsake.ai/docs/learn/analytics) for the environment variable naming precedence.

> HOMEBREW_NO_ANALYTICS 
> If set, do not send analytics. For more information, see: https://docs.brew.sh/Analytics

> export KEEPSAKE_NO_ANALYTICS=1

Author: By default we don't force, we just default to having it enabled. If it is disabled in config - DVC_ANALYTICS=true is a way to actually force it. I'm fine with adding NO_ANALYTICS too (got released already, I should've waited for the discussion :slightly_frowning_face: ) as an alias. Working on a PR.

Reviewer 2: It's still undocumented, so we could change if we want to.

Author: @skshetry Adjusted, please take a look. Thanks for the feedback! :pray:

This will disable it for the <abbr>project</abbr>. Alternatively, you can use
the `--global` or `--system` options of `dvc config` to disable analytics for
the active user or for everyone in the system, respectively.

Another way to disable analytics is to set the environment variable
`DVC_NO_ANALYTICS` (will override config).

> 

Reviewer: Guess not (it's failing checks).

```suggestion
```
